contracts/interfaces/ITwabRewards.sol

```diff
@@ -0,0 +1,170 @@
+        @param startTimestamp Timestamp at which the promotion starts
+        @param epochDuration Duration of one epoch in seconds
+        @param numberOfEpochs Number of epochs the promotion will last for
+     */
+    struct Promotion {
+        uint32 id;
+        bool cancelled;
+        address creator;
+        address ticket;
+        address token;
+        uint256 tokensPerEpoch;
+        uint32 startTimestamp;
+        uint32 epochDuration;
+        uint256 numberOfEpochs;
+    }
+
+    /**
+        @notice Parameters needed to create a promotion.
+        @param token Address of the token to be distributed
+        @param tokensPerEpoch Number of tokens to be distributed per epoch
+        @param startTimestamp Timestamp at which the promotion starts
+        @param epochDuration Duration of one epoch in seconds
+        @param numberOfEpochs Number of epochs the promotion will last for
+     */
+    struct PromotionParameters {
+        address token;
+        uint256 tokensPerEpoch;
+        uint32 startTimestamp;
+        uint32 epochDuration;
+        uint256 numberOfEpochs;
+    }
+
+    /**
+        @notice Create a new promotion.
+        @dev Will revert if a promotion is already active.
+        @dev For sake of simplicity, `msg.sender` will be the creator of the promotion.
+        @param _ticket Prize Pool ticket address for which the promotion is created
+        @param _promotionParameters Parameters needed to create a promotion
+        @return true if the creation was successful
+    */
+    function createPromotion(address _ticket, PromotionParameters calldata _promotionParameters)
+        external
+        returns (bool);
+
+    /**
+        @notice Cancel currently active promotion and send promotion tokens back to the creator.
+        @param _to Address that will receive the remaining tokens if there are any left.
+        @return true if cancelation was successful
+     */
+    function cancelPromotion(address _to) external returns (bool);
+
+    /**
+        @notice Extend promotion by adding more epochs.
+        @param _numberOfEpochs Number of epochs to add
+        @return true if the operation was successful
+     */
+    function extendPromotion(uint256 _numberOfEpochs) external returns (bool);
+
+    /**
+        @notice Claim rewards for a given promotion and epoch.
+        @dev Rewards can be claimed on behalf of a user.
+        @dev Caller may want to claim full or partial `amount` of rewards.
+        @param _user Address of the user to claim rewards for
+        @param _promotionId Promotion id to claim rewards for
+        @param _epochId Epoch id to claim rewards for
+        @param _amount Amount of tokens to claim
+        @return Amount of rewards claimed
+     */
+    function claimRewards(
+        address _user,
+        uint256 _promotionId,
+        uint256 _epochId,
+        uint256 _amount
+    ) external returns (uint256);
```

Comment:
Just a heads up: remember that a user can only claim **when the epoch is in the past**

Reply:
Done: https://github.com/pooltogether/v4-periphery/pull/8/commits/5e8563554515f6fdf14eb2629e35655351ccd91b